react/features/chat/components/native/ChatMessage.js

```diff
@@ -0,0 +1,152 @@
+// @flow
+
+import React from 'react';
+import { Text, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import { getLocalizedDateFormatter, translate } from '../../../base/i18n';
+import { Avatar } from '../../../base/participants';
+
+import AbstractChatMessage, {
+    _mapStateToProps as _abstractMapStateToProps,
+    type Props as AbstractProps
+} from '../AbstractChatMessage';
+import styles from './styles';
+
+/**
+ * Size of the rendered avatar in the message.
+ */
+const AVATAR_SIZE = 32;
+
+/**
+ * Formatter string to display the message timestamp.
+ */
+const TIMESTAMP_FORMAT = 'H:mm';
+
+type Props = AbstractProps & {
+
+    /**
+     * True if the chat window has a solid BG so then we have to adopt in style.
+     */
+    _solidBackground: boolean
+}
+
+/**
+ * Renders a single chat message.
+ */
+class ChatMessage extends AbstractChatMessage<Props> {
+    /**
+     * Implements {@code Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
```

Comment:
Overall I don't like this. It assumes the "iMessage" layout (sort of). But if we decide to use an approach more similar to Slack, then all messages are rendered the same. We should decide which way to go.